Sources/BraintreeCard/BTCardClient.m

```diff
@@ -133,19 +146,21 @@ - (void)tokenizeCard:(BTCard *)card completion:(void (^)(BTCardNonce *tokenizedC
                          }
                      }
                      
-                     [self sendAnalyticsEventWithSuccess:NO];
-
+                     [self.apiClient sendAnalyticsEvent:@"card:tokenize:failed"];
                      completion(nil, callbackError);
```

Comment:
Instead of sending this same event 9 times throughout the code, I think we should adopt a strategy like on PPCP. Example [here](https://github.com/paypal/iOS-SDK/blob/03b56a7e41855202d0f1183085af6f4e82df7598/Sources/CardPayments/CardClient.swift#L138-L141) - we have a helper function that includes sending the analytic event & calling the merchant's completion.

I think this will prevent us from over-sending events, and making sure we're only sending the corresponding event alongside the completion. 

I think a rule of thumb for the SDK of "each event string should only be searchable in the source code once" will help us out a lot / goes hand-in-hand with the above approach.